lib/api/controllers/writeController.js

```diff
@@ -205,6 +205,33 @@ function WriteController (kuzzle) {
         return kuzzle.pluginsManager.trigger('data:afterCreateCollection', new ResponseObject(modifiedRequestObject, response));
       });
   };
+
+  /**
+   * Validate a document against collection validation specifications.
+   *
+   * @param {RequestObject} requestObject
+   * @returns {Promise} promise resolved as a ResponseObject
+   */
+  this.validateDocument = function validateDocument (requestObject) {
+    var modifiedRequestObject = null;
+
+    return kuzzle.pluginsManager.trigger('data:beforeValidateDocument', requestObject)
+      .then(newRequestObject => {
+        modifiedRequestObject = newRequestObject;
+        return kuzzle.validation.validationPromise(modifiedRequestObject, true);
+      })
+      .then(response => {
+        var responseObject = new ResponseObject(modifiedRequestObject, response);
+
+        if (!response.validation) {
+          responseObject.status = 400;
+          responseObject.error = response.errorMessages;
+        }
+        kuzzle.pluginsManager.trigger('log:error', `The document does not comply with the ${requestObject.index} / ${requestObject.collection} : ${JSON.stringify(requestObject.data.body)}`);
```

Comment:
If there is a lot of activity, this can seriously hurt performances, along with clogging up log files. I think this should be silently rejected (except, of course, for a response to the user), and maybe we can add an event if someone want to listen to it (and configure the logger if one wants to)